drivers/include/net/netdev.h

```diff
@@ -215,6 +215,7 @@ extern "C" {
 enum {
     NETDEV_TYPE_UNKNOWN,
     NETDEV_TYPE_TEST,
+    NETDEV_TYPE_TEST_6LO,
```

Comment:
What is this needed for?

Reply:
To be able to use 6LoWPAN in the test, [`gnrc_netif_dev_is_6lo()`](https://github.com/maribu/RIOT/blob/263ce3df69905f1bb74de9343adc49d5f41c7e14/sys/net/gnrc/netif/gnrc_netif.c#L88-L108) has to return `true`. This is currently testing the netdev type, so just adding one special 6LO testing type was the easiest.

See https://github.com/RIOT-OS/RIOT/pull/15694/commits/9b62e61d2d8cd89fda8d170af2bc2dacc87de8d7

Reply:
I have to look more closely into your code (sadly no time atm :-(), but I would prefer if we could not have this. This seems like a bottleneck for scalability (on top of the bottleneck this struct already poses) when we start adding types for mock devices as well here. Can't you just give your test device e.g. the type `NETDEV_TYPE_IEEE802154` to have it behave as such? IIRC the `NETDEV_TYPE_TEST` is only for testing the `netdev_type`. Have a look e.g. at `tests/gnrc_netif` where I use "real" `netdev_type`s for mock devices.

Reply:
I can do so, but the test netdev needs to implement a lot more stuff to pass the self test during start up. This makes the test a bit more complex and increases ROM.

Reply:
Would be IMHO preferable to messing with the `netdev_type`s.

And a good invitation to provide generalized mock devices and/or interfaces for certain L2 protocols in the future so one does not have to re-implement them all the time. ^^